server/etcdserver/api/membership/store.go

```diff
@@ -220,8 +220,23 @@ func unsafeSaveMemberToStore(lg *zap.Logger, s v2store.Store, m *Member) error {
 		lg.Panic("failed to marshal raftAttributes", zap.Error(err))
 	}
 	p := path.Join(MemberStoreKey(m.ID), raftAttributesSuffix)
-	_, err = s.Create(p, false, string(b), false, v2store.TTLOptionSet{ExpireTime: v2store.Permanent})
-	return err
+	if _, err = s.Create(p, false, string(b), false, v2store.TTLOptionSet{ExpireTime: v2store.Permanent}); err != nil {
```

Comment:
This seems like a separate issue. Thanks for adding validation! However I'm worried about trying to fix two issues in one PR thus not putting enough attention into understanding each of them and introducing proper testing.

Reply:
> This seems like a separate issue.

Yes, it's a separate (but related) minor issue. When adding a new member, 
- v3store persists the whole member, 
- but v2store only persists the raftAttributes (peerURL, isLearner); so it causes a temprary inconsistency between v2 and v3store, and it depends on the following publish logic to update the attributes (name, clientURL), so it converge soon.

If we don't fix it in this PR, then the verification will fail, otherwise I will have to update the verification to ignore the attribute and only verify the raftAttributes(including peerURLs, isLearner).

Since from users perspective, this isn't an issue. We can also leave it as it's and only verify raftAttribute.

Reply:
> If we don't fix it in this PR, then the verification will fail, otherwise I will have to update the verification to ignore the attribute and only verify the raftAttributes(including peerURLs, isLearner).

That's why I'm suggesting to send PR to just fix this so we can merge it fast. This could also be backported to v3.4.

Reply:
> Since this isn't an issue from user perspective, so suggest to keep it as it's

I mean that I will revert the change.

Reply:
reverted the change.